src/js/profile/mobile/widget/DatePicker.js

```diff
@@ -0,0 +1,416 @@
+					content = document.createElement("div"),
+					dayPicker = self._buildDatePicker("day"),
+					monthPicker = self._buildDatePicker("month"),
+					yearPicker = self._buildDatePicker("year");
+
+				header.classList.add(DatePicker.classes.HEADER);
+				element.appendChild(header);
+
+				ui.header = header;
+
+				content.classList.add(DatePicker.classes.CONTENT);
+				content.appendChild(dayPicker);
+				content.appendChild(monthPicker);
+				content.appendChild(yearPicker);
+
+				element.appendChild(content);
+
+				return element;
+			};
+
+			/**
+			* Return Date value or value of one field
+			* @method _getValue
+			* @param {string} type
+			* @return {Date|number}
+			* @member ns.widget.mobile.DatePicker
+			* @protected
+			*/
+			prototype._getValue = function (type) {
+				var value = this._value;
+
+				switch (type) {
+					case "day":
+						return value.getDate();
+					case "month":
+						return value.getMonth() + 1;
+					case "year":
+						return value.getFullYear();
+					default:
+						return value;
+				}
+			}
+
+			/**
+			* Set the value of DatePicker
+			* @method _setValue
+			* @param {Date} value
+			* @member ns.widget.mobile.DatePicker
+			* @protected
+			*/
+			prototype._setValue = function (value) {
+				var self = this,
+					spins = self._spins,
+					headerText;
+
+				if (value instanceof Date) {
+					self._value = value;
+
+					Object.keys(spins).forEach(function (name) {
+						spins[name].value(self._getValue(name));
+					});
+
+					headerText = MONTH_NAMES[value.getMonth()] + " " + value.getFullYear();
+					self._ui.header.innerHTML = headerText;
+				}
+			};
+
+			/**
+			* Change a day value
+			* @method _changeDay
+			* @param {number} day
+			* @member ns.widget.mobile.DatePicker
+			* @protected
+			*/
+			prototype._changeDay = function (day) {
+				var self = this,
+					value = self.value();
+
+				value.setDate(day);
+				self.value(value);
+			};
+
+			/**
+			* Calculate the day count in month
+			* @method _dayInMonth
+			* @param {number} year
+			* @param {number} month
+			* @return {number}
+			* @member ns.widget.mobile.DatePicker
+			* @protected
+			*/
+			prototype._dayInMonth = function (year, month) {
+				if (year === undefined) {
+					year = this._getValue("year");
+				}
+				if (month === undefined) {
+					month = this._getValue("month");
+				}
+				return new Date(year, month, 0).getDate();
+			}
+
+			/**
+			* Change a month value
+			* @method _changeMonth
+			* @param {number} month
+			* @member ns.widget.mobile.DatePicker
+			* @protected
+			*/
+			prototype._changeMonth = function (month) {
+				var self = this,
+					spins = self._spins,
+					value = self.value(),
+					day = value.getDate(),
+					dayInMonth;
+
+				dayInMonth = self._dayInMonth(value.getFullYear(), month);
+
+				if (day > dayInMonth) {
+					day = dayInMonth;
+				}
+
+				value.setDate(day);
+				value.setMonth(month - 1);
+
+				spins.day.option("max", dayInMonth);
```

Comment:
Why do you need spins.day.option("max", dayInMonth)?

Reply:
Need to set the max when changing the month.
Ex) 1. set the 31/Mar
       2. Change the Month to the "Feb"
       3. Should change the "31" to the "28" or "29".